lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
```

Comment:
It seems to be idiomatic to call a struct type `t`. I understand the use of `state` as attempting to give this an intention revealing name, but I thought it would be worth considering.

Should our state really just be a port instead of a single key struct? The pull request is about types that are currently used, but what can we learn from looking at the types in the system? With a struct, we can't add or remove keys so our state really is just a port. Is there a benefit to having a struct other than adding a name to it?